lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8730,13 +8728,12 @@ SourceFile &ClangImporter::Implementation::getClangSwiftAttrSourceFile(
   // Note that this is for an attribute.
   sourceMgr.setGeneratedSourceInfo(
       bufferID,
-      {
-        GeneratedSourceInfo::AttributeFromClang,
-        CharSourceRange(),
-        sourceMgr.getRangeForBuffer(bufferID),
-        &module
-      }
-  );
+      {GeneratedSourceInfo::AttributeFromClang,
+       // Below relies on originalRange residing in the same SourceManager
+       // as SwiftContext.SourceMgr, which is guaranteed via the initialization
```

Comment:
For my own edification, are not all `swift::SourceRange`s residing with The One `swift::SourceManager`? I get confused with all the `clang::SourceRange`, `clang::SourceManager` etc. thrown in the mix. Or are there multiple `swift::SourceManager`s floating around in the same process?

Reply:
So from my understanding, there should only be one `swift::SourceManager` instance, but are so many references scattered throughout the ClangImporter subsystem that I'm not entirely sure (I _think_ those are all there for convenience but I don't know that for sure).

`clang::SourceManager` is a distinct thing that manages source locations and ranges coming from Clang. `ClangSourceBufferImporter` does (some of?) the bridging for us between the two source managers.